### PR TITLE
Fuzzing: Add 2 collations fuzzers

### PR DESCRIPTION
**Pull request by AdamKorcz**
## Description

Adds two collations fuzzers and instructs OSS-fuzz to include them.

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/fuzzing/collate_fuzzer.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+package fuzzing
+
+import (
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+
+	"vitess.io/vitess/go/mysql/collations"
+)
+
+func FuzzCollateUnicode(data []byte) int {
+	c := &collations.Collation_unicode_general_ci{}
```
Comment (vmg) [inline go/test/fuzzing/collate_fuzzer.go:23]: This is not correct, you're not supposed to instantiate collation objects directly: they must be looked up directly from an Environment. I guess this is not properly explained in the API docs. :)

File: go/test/fuzzing/collate_fuzzer.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	isPrefix, err := f.GetBool()
+	if err != nil {
+		return 0
+	}
+	_ = c.Collate(left, right, isPrefix)
+	return 1
+}
+
+func FuzzCollateWildcard(data []byte) int {
+	c := &collations.Collation_multibyte{}
```
Comment (vmg) [inline go/test/fuzzing/collate_fuzzer.go:46]: Likewise, this is not a valid collation.

File: go/test/fuzzing/collate_fuzzer.go (hunk 1 above)
Comment (AdamKorcz) [reply go/test/fuzzing/collate_fuzzer.go]: Thanks for the heads up. I rewrote it a bit to initialize the same way as [the existing fuzzer](https://github.com/vitessio/vitess/blob/main/go/mysql/collations/uca_test.go#L53).

File: go/mysql/collations/fuzz.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	left, err := f.GetBytes()
+	if err != nil {
+		return 0
+	}
+	right, err := f.GetBytes()
+	if err != nil {
+		return 0
+	}
+	_ = coll.Collate(left, right, false)
+	return 1
```
Comment (frouioui) [inline go/mysql/collations/fuzz.go:51]: are we asserting anything or simply making sure it does not crash?

Comment (systay) [reply go/mysql/collations/fuzz.go:51]: fuzzing is all about the latter. we are just making sure that nothing panics

**Review by vmg — APPROVED** [review]

**Comment by AdamKorcz** [comment]
This one alone with https://github.com/vitessio/vitess/pull/9316 and https://github.com/vitessio/vitess/pull/9339 should be ready.

**Comment by systay** [comment]
Replaced by #9475

**Pull request closed without merging**